src/vuestic-theme/vuestic-components/va-slider/VaSlider.vue

```diff
@@ -0,0 +1,582 @@
+        this.setTransform()
+        this.setCurrentValue(valueRange[0])
+      } else {
+        this.setTransform()
+        this.setCurrentValue(valueRange[1])
+      }
+    },
+    setValue (val) {
+      val = this.limitValue(val)
+
+      if (this.isDiff(this.val, val)) {
+        this.val = val
+      }
+    },
+    setTransform () {
+      if (this.isRange) {
+        const slider = this.currentSlider,
+          val0 = ((this.value[0] - this.min) / (this.max - this.min)) * 100,
+          val1 = ((this.value[1] - this.min) / (this.max - this.min)) * 100,
+          processSize = `${val1 - val0}%`,
+          processPos = `${val0}%`
+
+        this.$refs.process.style.width = processSize
+        this.$refs.process.style['left'] = processPos
+        if (slider === 0) {
+          this.$refs.dot0.style['left'] = `calc('${processPos} - 8px)`
+        } else {
+          this.$refs.dot1.style['left'] = `calc('${processPos} - 8px)`
+        }
+      } else {
+        const val = ((this.value - this.min) / (this.max - this.min)) * 100
+
+        this.$refs.process.style.width = `${val}%`
+        this.$refs.dot.style['left'] = `calc('${val} - 8px)`
+      }
+    },
+    normalizeValue (value) {
+      let currentRest = value % this.step
+      if ((currentRest / this.step) >= 0.5) {
+        value = value + (this.step - currentRest)
+      } else {
+        value = value - currentRest
+      }
+      return value
+    },
+    limitValue (val) {
+      const inRange = (v) => {
+        if (v < this.min) {
+          return this.min
+        } else if (v > this.max) {
+          return this.max
+        }
+        return v
+      }
+
+      if (this.isRange) {
+        if (val[0] >= val[1] && this.currentSlider === 0) {
+          return [val[1], val[1]]
+        }
+        if (val[0] >= val[1] && this.currentSlider === 1) {
+          return [val[0], val[0]]
+        }
+        return val.map((v) => inRange(v))
+      } else {
+        return inRange(val)
+      }
+    },
+    isDiff (a, b) {
+      if (Object.prototype.toString.call(a) !== Object.prototype.toString.call(b)) {
+        return true
+      } else if (Array.isArray(a) && a.length === b.length) {
+        return a.some((v, i) => v !== b[i])
+      }
+      return a !== b
+    },
+    validateBorders () {
+      let isRightBorders = true
+
+      if (this.max < this.min) {
+        console.error('The maximum value can not be less than the minimum value.')
+        isRightBorders = false
+      }
+
+      if (this.min > this.max) {
+        console.error('The minimum value can not be greater than the maximum value.')
+        isRightBorders = false
+      }
+
+      return isRightBorders
+    },
+    validateValue (val) {
+      let isRightValue = true
+
+      const inRange = (v) => {
+        if (v < this.min) {
+          console.error(`The value of the slider is ${v}, the minimum value is ${this.min}, the value of this slider can not be less than the minimum value`)
```

Comment:
throw new Error. it is a bad practice to use cl for error displaying. And maybe it better to round the value to minimum/ maximum in this case